src/diff/children.js

```diff
@@ -304,10 +310,8 @@ export function toChildArray(children, out) {
 }
 
 function placeChild(parentDom, newDom, oldDom) {
-	if (oldDom == null || oldDom.parentNode !== parentDom) {
-		parentDom.insertBefore(newDom, null);
-	} else if (newDom != oldDom || newDom.parentNode == null) {
```

Comment:
The `newDom.parentNode == null` condition can also be removed cuz it would never happen, or if did would result in a DOMException.

For this condition to be hit, `newDom` would have to be equal to `oldDom` cuz of the check prior to this one. So in that scenario, if `newDom.parentNode == null`, then `oldDom.parentNode == null`. If oldDom is unmounted (since it has no parentNode), then calling `parentDom.insertBefore(newDom, oldDom)` would error cuz oldDom is unmounted and you can't insert a node before another node that is unmounted.